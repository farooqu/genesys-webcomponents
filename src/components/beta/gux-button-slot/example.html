<h1>gux-button-slot-beta</h1>

<h2>Slotted Button Tag</h2>

<h3>Enabled</h3>

<div>
  <gux-button-slot-beta>
    <button type="button" onclick="notify(event)">Default</button>
  </gux-button-slot-beta>

  <gux-button-slot-beta accent="primary">
    <button type="button" onclick="notify(event)">Primary</button>
  </gux-button-slot-beta>

  <gux-button-slot-beta accent="secondary">
    <button type="button" onclick="notify(event)">Secondary</button>
  </gux-button-slot-beta>

  <gux-button-slot-beta accent="tertiary">
    <button type="button" onclick="notify(event)">Tertiary</button>
  </gux-button-slot-beta>

  <gux-button-slot-beta>
    <button type="button" onclick="notify(event)">
      <span>Text in a span element</span>
    </button>
  </gux-button-slot-beta>

  <gux-button-slot-beta>
    <button type="button" onclick="notify(event)">
      <span>Text in a span element</span>
      <span>More text in a span element</span>
    </button>
  </gux-button-slot-beta>

  <gux-button-slot-beta>
    <button type="button" onclick="notify(event)">
      <gux-icon
        icon-name="trashcan"
        screenreader-text="This will be read by a screen reader"
      ></gux-icon>
    </button>
  </gux-button-slot-beta>

  <gux-button-slot-beta>
    <button
      type="button"
      title="This will be read by a screen reader"
      onclick="notify(event)"
    >
      <gux-icon icon-name="trashcan" decorative></gux-icon>
    </button>
  </gux-button-slot-beta>

  <gux-button-slot-beta>
    <button type="button" onclick="notify(event)">
      <gux-icon icon-name="trashcan" decorative></gux-icon>
      <span>Text in a span element</span>
    </button>
  </gux-button-slot-beta>
</div>

<h3>Disabled</h3>

<div>
  <gux-button-slot-beta>
    <button type="button" disabled onclick="notify(event)">Default</button>
  </gux-button-slot-beta>

  <gux-button-slot-beta accent="primary">
    <button type="button" disabled onclick="notify(event)">Primary</button>
  </gux-button-slot-beta>

  <gux-button-slot-beta accent="secondary">
    <button type="button" disabled onclick="notify(event)">Secondary</button>
  </gux-button-slot-beta>

  <gux-button-slot-beta accent="tertiary">
    <button type="button" disabled onclick="notify(event)">Tertiary</button>
  </gux-button-slot-beta>

  <gux-button-slot-beta>
    <button type="button" disabled onclick="notify(event)">
      <span>Text in a span element</span>
    </button>
  </gux-button-slot-beta>

  <gux-button-slot-beta>
    <button type="button" disabled onclick="notify(event)">
      <span>Text in a span element</span>
      <span>More text in a span element</span>
    </button>
  </gux-button-slot-beta>

  <gux-button-slot-beta>
    <button type="button" disabled onclick="notify(event)">
      <gux-icon
        icon-name="trashcan"
        screenreader-text="This will be read by a screen reader"
      ></gux-icon>
    </button>
  </gux-button-slot-beta>

  <gux-button-slot-beta>
    <button
      type="button"
      disabled
      title="This will be read by a screen reader"
      onclick="notify(event)"
    >
      <gux-icon icon-name="trashcan" decorative></gux-icon>
    </button>
  </gux-button-slot-beta>

  <gux-button-slot-beta>
    <button type="button" disabled onclick="notify(event)">
      <gux-icon icon-name="trashcan" decorative></gux-icon>
      <span>Text in a span element</span>
    </button>
  </gux-button-slot-beta>
</div>

<h2>Slotted Input Tag (button)</h2>

<h3>Enabled</h3>

<div>
  <gux-button-slot-beta>
    <input type="button" onclick="notify(event)" value="Default" />
  </gux-button-slot-beta>

  <gux-button-slot-beta accent="primary">
    <input type="button" onclick="notify(event)" value="Primary" />
  </gux-button-slot-beta>

  <gux-button-slot-beta accent="secondary">
    <input type="button" onclick="notify(event)" value="Secondary" />
  </gux-button-slot-beta>

  <gux-button-slot-beta accent="tertiary">
    <input type="button" onclick="notify(event)" value="Tertiary" />
  </gux-button-slot-beta>
</div>

<h3>Disabled</h3>

<div>
  <gux-button-slot-beta>
    <input type="button" disabled onclick="notify(event)" value="Default" />
  </gux-button-slot-beta>

  <gux-button-slot-beta accent="primary">
    <input type="button" disabled onclick="notify(event)" value="Primary" />
  </gux-button-slot-beta>

  <gux-button-slot-beta accent="secondary">
    <input type="button" disabled onclick="notify(event)" value="Secondary" />
  </gux-button-slot-beta>

  <gux-button-slot-beta accent="tertiary">
    <input type="button" disabled onclick="notify(event)" value="Tertiary" />
  </gux-button-slot-beta>
</div>

<h2>Slotted Input Tag (submit)</h2>

<h3>Enabled</h3>

<div>
  <gux-button-slot-beta>
    <input type="submit" onclick="notify(event)" value="Default" />
  </gux-button-slot-beta>

  <gux-button-slot-beta accent="primary">
    <input type="submit" onclick="notify(event)" value="Primary" />
  </gux-button-slot-beta>

  <gux-button-slot-beta accent="secondary">
    <input type="submit" onclick="notify(event)" value="Secondary" />
  </gux-button-slot-beta>

  <gux-button-slot-beta accent="tertiary">
    <input type="submit" onclick="notify(event)" value="Tertiary" />
  </gux-button-slot-beta>
</div>

<h3>Disabled</h3>

<div>
  <gux-button-slot-beta>
    <input type="submit" disabled onclick="notify(event)" value="Default" />
  </gux-button-slot-beta>

  <gux-button-slot-beta accent="primary">
    <input type="submit" disabled onclick="notify(event)" value="Primary" />
  </gux-button-slot-beta>

  <gux-button-slot-beta accent="secondary">
    <input type="submit" disabled onclick="notify(event)" value="Secondary" />
  </gux-button-slot-beta>

  <gux-button-slot-beta accent="tertiary">
    <input type="submit" disabled onclick="notify(event)" value="Tertiary" />
  </gux-button-slot-beta>
</div>

<style>
  gux-button-slot-beta {
    margin: 10px;
  }
</style>
