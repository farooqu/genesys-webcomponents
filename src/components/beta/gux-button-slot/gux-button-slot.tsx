import { Component, Element, h, Host, JSX, Prop } from '@stencil/core';

import { trackComponent } from '../../../usage-tracking';
import { logError } from '../../../utils/error/log-error';
import { GuxButtonAccent } from '../../stable/gux-button/gux-button.types';

/**
 * @slot - button, input[type="button"] or input[type="submit"] element
 */
@Component({
  styleUrl: 'gux-button-slot.less',
  tag: 'gux-button-slot-beta',
  shadow: true
})
export class GuxButtonSlot {
  @Element() root: HTMLElement;

  @Prop()
  accent: GuxButtonAccent = 'secondary';

  private validateSlotContent(): void {
    const slottedElement = this.root.children[0];
    const slottedTagName = slottedElement.tagName;

    if (slottedTagName === 'BUTTON') {
      return;
    } else if (slottedTagName === 'INPUT') {
      const slottedType = slottedElement.getAttribute('type');

      if (slottedType === 'button' || slottedType === 'submit') {
        return;
      }
    }

    logError(
      'gux-button-slot',
      'You must slot a button, input[type="button"] or input[type="submit"] element.'
    );
  }

  componentWillLoad(): void {
    trackComponent(this.root);

    this.validateSlotContent();
  }

  render(): JSX.Element {
    return (
      <Host accent={this.accent}>
        <slot />
      </Host>
    ) as JSX.Element;
  }
}
