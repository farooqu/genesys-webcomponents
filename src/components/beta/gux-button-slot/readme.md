# gux-button-slot
<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description | Type                                     | Default       |
| -------- | --------- | ----------- | ---------------------------------------- | ------------- |
| `accent` | `accent`  |             | `"primary" \| "secondary" \| "tertiary"` | `'secondary'` |


## Slots

| Slot | Description                                                  |
| ---- | ------------------------------------------------------------ |
|      | button, input[type="button"] or input[type="submit"] element |


## Dependencies

### Used by

 - [gux-action-button](../../stable/gux-action-button)
 - [gux-button-multi](../gux-button-multi)
 - [gux-pagination-buttons](../../stable/gux-pagination/gux-pagination-buttons)
 - [gux-pagination-cursor](../../stable/gux-pagination-cursor)
 - [gux-table-beta](../gux-table)

### Graph
```mermaid
graph TD;
  gux-action-button --> gux-button-slot-beta
  gux-button-multi --> gux-button-slot-beta
  gux-pagination-buttons --> gux-button-slot-beta
  gux-pagination-cursor --> gux-button-slot-beta
  gux-table-beta --> gux-button-slot-beta
  style gux-button-slot-beta fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
