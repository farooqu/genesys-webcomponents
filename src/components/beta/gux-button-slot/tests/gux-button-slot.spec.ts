jest.mock('../../../../utils/error/log-error', () => ({
  __esModule: true,
  logError: jest.fn()
}));

import { newSpecPage } from '@stencil/core/testing';

import { GuxButtonSlot } from '../gux-button-slot';

import { logError } from '../../../../utils/error/log-error';

const components = [GuxButtonSlot];
const language = 'en';

describe('gux-button-slot-beta', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('#render', () => {
    [
      {
        description: 'should render default button',
        html: '<gux-button-slot-beta><button type="button"accent="primary">Button</button></gux-button-slot-beta>'
      },
      {
        description: 'should render primary button',
        html: '<gux-button-slot-beta accent="primary"><button type="button"accent="primary">Button</button></gux-button-slot-beta>'
      },
      {
        description: 'should render secondary button',
        html: '<gux-button-slot-beta accent="secondary"><button type="button"accent="primary">Button</button></gux-button-slot-beta>'
      },
      {
        description: 'should render tertiary button',
        html: '<gux-button-slot-beta accent="tertiary"><button type="button"accent="primary">Button</button></gux-button-slot-beta>'
      },
      {
        description: 'should render invalid button',
        html: '<gux-button-slot-beta accent="invalid"><button type="button"accent="primary">Button</button></gux-button-slot-beta>'
      },
      {
        description: 'should render disabled default button',
        html: '<gux-button-slot-beta><button type="button"accent="primary" disabled>Button</button></gux-button-slot-beta>'
      },
      {
        description: 'should render disabled primary button',
        html: '<gux-button-slot-beta accent="primary"><button type="button"accent="primary" disabled>Button</button></gux-button-slot-beta>'
      },
      {
        description: 'should render disabled secondary button',
        html: '<gux-button-slot-beta accent="secondary"><button type="button"accent="primary" disabled>Button</button></gux-button-slot-beta>'
      },
      {
        description: 'should render disabled tertiary button',
        html: '<gux-button-slot-beta accent="tertiary"><button type="button"accent="primary" disabled>Button</button></gux-button-slot-beta>'
      },
      {
        description: 'should render button with invalid accent',
        html: '<gux-button-slot-beta accent="invalid"><button type="button"accent="primary">Button</button></gux-button-slot-beta>'
      },
      {
        description: 'should render default input[type=button]',
        html: '<gux-button-slot-beta><input type="button"accent="primary" value="Button"/></gux-button-slot-beta>'
      },
      {
        description: 'should render primary input[type=button]',
        html: '<gux-button-slot-beta accent="primary"><input type="button"accent="primary" value="Button"/></gux-button-slot-beta>'
      },
      {
        description: 'should render secondary input[type=button]',
        html: '<gux-button-slot-beta accent="secondary"><input type="button"accent="primary" value="Button"/></gux-button-slot-beta>'
      },
      {
        description: 'should render tertiary input[type=button]',
        html: '<gux-button-slot-beta accent="tertiary"><input type="button"accent="primary" value="Button"/></gux-button-slot-beta>'
      },
      {
        description: 'should render invalid input[type=button]',
        html: '<gux-button-slot-beta accent="invalid"><input type="button"accent="primary" value="Button"/></gux-button-slot-beta>'
      },
      {
        description: 'should render disabled default input[type=button]',
        html: '<gux-button-slot-beta><input type="button"accent="primary" disabled value="Button"/></gux-button-slot-beta>'
      },
      {
        description: 'should render disabled primary input[type=button]',
        html: '<gux-button-slot-beta accent="primary"><input type="button"accent="primary" disabled value="Button"/></gux-button-slot-beta>'
      },
      {
        description: 'should render disabled secondary input[type=button]',
        html: '<gux-button-slot-beta accent="secondary"><input type="button"accent="primary" disabled value="Button"/></gux-button-slot-beta>'
      },
      {
        description: 'should render disabled tertiary input[type=button]',
        html: '<gux-button-slot-beta accent="tertiary"><input type="button"accent="primary" disabled value="Button"/></gux-button-slot-beta>'
      },
      {
        description: 'should render input[type=button] with invalid accent',
        html: '<gux-button-slot-beta accent="invalid"><input type="button"accent="primary" value="Button"/></gux-button-slot-beta>'
      },
      {
        description: 'should render default input[type=submit]',
        html: '<gux-button-slot-beta><input type="submit"accent="primary" value="Button"/></gux-button-slot-beta>'
      },
      {
        description: 'should render primary input[type=submit]',
        html: '<gux-button-slot-beta accent="primary"><input type="submit"accent="primary" value="Button"/></gux-button-slot-beta>'
      },
      {
        description: 'should render secondary input[type=submit]',
        html: '<gux-button-slot-beta accent="secondary"><input type="submit"accent="primary" value="Button"/></gux-button-slot-beta>'
      },
      {
        description: 'should render tertiary input[type=submit]',
        html: '<gux-button-slot-beta accent="tertiary"><input type="submit"accent="primary" value="Button"/></gux-button-slot-beta>'
      },
      {
        description: 'should render invalid input[type=submit]',
        html: '<gux-button-slot-beta accent="invalid"><input type="submit"accent="primary" value="Button"/></gux-button-slot-beta>'
      },
      {
        description: 'should render disabled default input[type=submit]',
        html: '<gux-button-slot-beta><input type="submit"accent="primary" disabled value="Button"/></gux-button-slot-beta>'
      },
      {
        description: 'should render disabled primary input[type=submit]',
        html: '<gux-button-slot-beta accent="primary"><input type="submit"accent="primary" disabled value="Button"/></gux-button-slot-beta>'
      },
      {
        description: 'should render disabled secondary input[type=submit]',
        html: '<gux-button-slot-beta accent="secondary"><input type="submit"accent="primary" disabled value="Button"/></gux-button-slot-beta>'
      },
      {
        description: 'should render disabled tertiary input[type=submit]',
        html: '<gux-button-slot-beta accent="tertiary"><input type="submit"accent="primary" disabled value="Button"/></gux-button-slot-beta>'
      },
      {
        description: 'should render input[type=submit] with invalid accent',
        html: '<gux-button-slot-beta accent="invalid"><input type="submit"accent="primary" value="Button"/></gux-button-slot-beta>'
      }
    ].forEach(({ description, html }) => {
      it(description, async () => {
        const page = await newSpecPage({ components, html, language });

        expect(page.rootInstance).toBeInstanceOf(GuxButtonSlot);
        expect(page.root).toMatchSnapshot();
      });
    });
  });

  describe('slot content', () => {
    it('should not log an error if a button element is slotted', async () => {
      await newSpecPage({
        components,
        html: '<gux-button-slot-beta><button type="button"accent="primary">Button</button></gux-button-slot-beta>',
        language
      });

      expect(logError).not.toHaveBeenCalled();
    });

    it('should not log an error if an input[type=button] element is slotted', async () => {
      await newSpecPage({
        components,
        html: '<gux-button-slot-beta><input type="button"accent="primary" value="Button"/></gux-button-slot-beta>',
        language
      });

      expect(logError).not.toHaveBeenCalled();
    });

    it('should not log an error if an input[type=submit] element is slotted', async () => {
      await newSpecPage({
        components,
        html: '<gux-button-slot-beta><input type="submit"accent="primary" value="Button"/></gux-button-slot-beta>',
        language
      });

      expect(logError).not.toHaveBeenCalled();
    });

    it('should not log an error if an input[type=text] element is slotted', async () => {
      await newSpecPage({
        components,
        html: '<gux-button-slot-beta><input type="text"accent="primary" value="Button"/></gux-button-slot-beta>',
        language
      });

      expect(logError).toHaveBeenCalledWith(
        'gux-button-slot',
        'You must slot a button, input[type="button"] or input[type="submit"] element.'
      );
    });

    it('should not log an error if a button element is slotted', async () => {
      await newSpecPage({
        components,
        html: '<gux-button-slot-beta><div>Not a button</div></gux-button-slot-beta>',
        language
      });

      expect(logError).toHaveBeenCalledWith(
        'gux-button-slot',
        'You must slot a button, input[type="button"] or input[type="submit"] element.'
      );
    });
  });
});
