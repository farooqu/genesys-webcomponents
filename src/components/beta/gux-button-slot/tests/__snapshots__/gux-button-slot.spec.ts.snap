// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-button-slot-beta #render should render button with invalid accent 1`] = `
<gux-button-slot-beta accent="invalid">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <button accent="primary" type="button">
    Button
  </button>
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render default button 1`] = `
<gux-button-slot-beta accent="secondary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <button accent="primary" type="button">
    Button
  </button>
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render default input[type=button] 1`] = `
<gux-button-slot-beta accent="secondary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <input accent="primary" type="button" value="Button">
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render default input[type=submit] 1`] = `
<gux-button-slot-beta accent="secondary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <input accent="primary" type="submit" value="Button">
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render disabled default button 1`] = `
<gux-button-slot-beta accent="secondary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <button accent="primary" disabled="" type="button">
    Button
  </button>
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render disabled default input[type=button] 1`] = `
<gux-button-slot-beta accent="secondary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <input accent="primary" disabled="" type="button" value="Button">
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render disabled default input[type=submit] 1`] = `
<gux-button-slot-beta accent="secondary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <input accent="primary" disabled="" type="submit" value="Button">
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render disabled primary button 1`] = `
<gux-button-slot-beta accent="primary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <button accent="primary" disabled="" type="button">
    Button
  </button>
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render disabled primary input[type=button] 1`] = `
<gux-button-slot-beta accent="primary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <input accent="primary" disabled="" type="button" value="Button">
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render disabled primary input[type=submit] 1`] = `
<gux-button-slot-beta accent="primary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <input accent="primary" disabled="" type="submit" value="Button">
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render disabled secondary button 1`] = `
<gux-button-slot-beta accent="secondary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <button accent="primary" disabled="" type="button">
    Button
  </button>
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render disabled secondary input[type=button] 1`] = `
<gux-button-slot-beta accent="secondary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <input accent="primary" disabled="" type="button" value="Button">
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render disabled secondary input[type=submit] 1`] = `
<gux-button-slot-beta accent="secondary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <input accent="primary" disabled="" type="submit" value="Button">
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render disabled tertiary button 1`] = `
<gux-button-slot-beta accent="tertiary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <button accent="primary" disabled="" type="button">
    Button
  </button>
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render disabled tertiary input[type=button] 1`] = `
<gux-button-slot-beta accent="tertiary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <input accent="primary" disabled="" type="button" value="Button">
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render disabled tertiary input[type=submit] 1`] = `
<gux-button-slot-beta accent="tertiary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <input accent="primary" disabled="" type="submit" value="Button">
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render input[type=button] with invalid accent 1`] = `
<gux-button-slot-beta accent="invalid">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <input accent="primary" type="button" value="Button">
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render input[type=submit] with invalid accent 1`] = `
<gux-button-slot-beta accent="invalid">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <input accent="primary" type="submit" value="Button">
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render invalid button 1`] = `
<gux-button-slot-beta accent="invalid">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <button accent="primary" type="button">
    Button
  </button>
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render invalid input[type=button] 1`] = `
<gux-button-slot-beta accent="invalid">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <input accent="primary" type="button" value="Button">
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render invalid input[type=submit] 1`] = `
<gux-button-slot-beta accent="invalid">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <input accent="primary" type="submit" value="Button">
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render primary button 1`] = `
<gux-button-slot-beta accent="primary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <button accent="primary" type="button">
    Button
  </button>
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render primary input[type=button] 1`] = `
<gux-button-slot-beta accent="primary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <input accent="primary" type="button" value="Button">
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render primary input[type=submit] 1`] = `
<gux-button-slot-beta accent="primary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <input accent="primary" type="submit" value="Button">
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render secondary button 1`] = `
<gux-button-slot-beta accent="secondary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <button accent="primary" type="button">
    Button
  </button>
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render secondary input[type=button] 1`] = `
<gux-button-slot-beta accent="secondary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <input accent="primary" type="button" value="Button">
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render secondary input[type=submit] 1`] = `
<gux-button-slot-beta accent="secondary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <input accent="primary" type="submit" value="Button">
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render tertiary button 1`] = `
<gux-button-slot-beta accent="tertiary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <button accent="primary" type="button">
    Button
  </button>
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render tertiary input[type=button] 1`] = `
<gux-button-slot-beta accent="tertiary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <input accent="primary" type="button" value="Button">
</gux-button-slot-beta>
`;

exports[`gux-button-slot-beta #render should render tertiary input[type=submit] 1`] = `
<gux-button-slot-beta accent="tertiary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <input accent="primary" type="submit" value="Button">
</gux-button-slot-beta>
`;
